contracts/hardhat.config.ts

```diff
@@ -18,6 +20,14 @@ const COMPILER_SETTINGS = {
   },
 }
 
+// prune forge style tests from hardhat paths
+subtask(TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS).setAction(
+  async (_, __, runSuper) => {
+    const paths = await runSuper()
+    return paths.filter((p: string) => !p.endsWith('.t.sol'))
```

Comment:
Just to be sure, is this the only way we can differentiate foundry vs hardhat tests? If it's not too difficult, I would also consider filtering by directory names. If not though, I think this is fine.